src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -20,9 +20,9 @@ internal sealed partial class ThreadPoolWorkQueue
     {
         internal static class WorkStealingQueueList
         {
-#pragma warning disable CA1825 // avoid the extra generic instantiation for Array.Empty<T>(); this is the only place we'll ever create this array
+#pragma warning disable CA1825, IDE0300 // avoid the extra generic instantiation for Array.Empty<T>(); this is the only place we'll ever create this array
```

Comment:
Wonder if this is a case where it'd be beneficial for NAOT and/or the JIT to special case `Array.Empty<T>` and allow avoiding the generic instantiation so that this isn't a place we need to special case everywhere

Reply:
cc: @jkotas

Reply:
It would require one-off special casing in number of places throughout the runtimes and the special casing would still be less efficient (more fixed runtime overhead) compared to disabling the warning in cases like this one when Array.Empty is not helping.

Reply:
> this isn't a place we need to special case everywhere

This special casing and disabling the warning is important for generic types list `List<T>` where the cost of Array.Empty instantiation gets multiplied and where Array.Empty has other inefficiencies (e.g. extra generic dictionary lookups in shared generic code).

Non-generic cases are typically just micro-optimizations. I would not have a problem with switching them to Array.Empty if it is just a static footprint microoptimization. We generally do not think twice about small impacts of canonical C# on static footprint.

For this specific case, it is not obvious to me whether this is just a static footprint micro-optimization or whether it is also important for throughput (avoiding cctor checks) in some cases.